### ISSUE TITLE
Allow `as` for `no-dynamic-keys` rule

### ISSUE THREAD

**Issue by shaedrich** [intlify/eslint-plugin-vue-i18n#602]
**What rule do you want to change?**
[`@intlify/vue-i18n/no-dynamic-keys`](https://github.com/intlify/eslint-plugin-vue-i18n/blob/master/lib/rules/no-dynamic-keys.ts)

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option ([`skipTSAsExpression`](https://github.com/intlify/eslint-plugin-vue-i18n/blob/master/lib/utils/index.ts#L525-L541))

**Please provide some example code that this change will affect:**

```vue
<template>
<span>{{ t(`binary-question.${isPositive.toString()}.title` as `binary-question.${'true' | 'false'}.title` }}</span>
</template>

<script lang="ts" setup>
let isPositive: bool = true;
</script>
```

**What does the rule currently do for this code?**
Disallow _any_ dynamic translation key

**What will the rule do after it's changed?**
Only disallow dynamic translation keys when their type isn't narrowed down via template string literal type.

**Additional context**
